clients/privacy-center/common/fides-user-cookie.ts

```diff
@@ -0,0 +1,56 @@
+import { getCookie, setCookie, Types } from "typescript-cookie";
+
+export const FIDES_USER_DEVICE_ID_COOKIE_NAME = "fides_user_device_id";
+const MAX_AGE_DAYS = 1;
+
+const CODEC: Types.CookieCodecConfig<string, string> = {
+  decodeName: decodeURIComponent,
+  decodeValue: decodeURIComponent,
+  encodeName: encodeURIComponent,
+  encodeValue: encodeURIComponent,
+};
+/**
+ * Generates a new `fides_user_device_id` and stores it to the cookie
+ */
+const setFidesUserDeviceIdCookie = () => {
+  if (typeof document === "undefined") {
+    return undefined;
+  }
+
+  const rootDomain = window.location.hostname.split(".").slice(-2).join(".");
+  const uuid = crypto.randomUUID();
+
+  setCookie(
+    FIDES_USER_DEVICE_ID_COOKIE_NAME,
+    uuid,
+    {
+      domain: rootDomain,
+      expires: MAX_AGE_DAYS,
+    },
+    CODEC
+  );
+  return uuid;
+};
+
+/**
+ * Returns the `fides_user_device_id` from the cookie
+ */
+const getFidesUserDeviceIdCookie = () => {
+  if (typeof document === "undefined") {
+    return undefined;
+  }
+
+  return getCookie(FIDES_USER_DEVICE_ID_COOKIE_NAME, CODEC);
+};
+
+/**
+ * Retrieves the fides user device UUID as stored in the `fides_user_device_id` cookie
+ * If no cookie is found, sets a new UUID on the cookie and returns the value
+ */
+export const getFidesUserDeviceUuid = () => {
```

Comment:
fyi I renamed this—again, I know I was the one who suggested to make this a hook with `use...`, but it's not actually a hook since it doesn't use react state or anything. it's just a plain ol' function, so best not to prepend it with `use` which implies a hook (and implies you can only use it at the beginning of a component, which isn't true and would be very limiting!) 

that said, I don't love this name since it gets but also sets if it can't get. there must be a word for that, right... it's escaping me at the moment 😄